pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -94,83 +154,195 @@ private FunctionRegistry() {
   public static void init() {
   }
 
+  @VisibleForTesting
+  public static void registerFunction(Method method) {
+    registerFunction(method, Collections.singleton(method.getName()), false, false);
```

Comment:
When this function will be called? Specifically, this method seems to be not thread safe. If some thread is registering a function and another is looking for functions, the behavior is not specified. Is that something that can happen? It looks like the method is public for testing proposes, so I would assume the production code won't call this method but:
- We should specify that in the Javadoc of this method
- Does it mean we cannot run tests concurrently?

Reply:
this is only for testing.